compiler/env/OMRAotAdapter.cpp

```diff
@@ -0,0 +1,142 @@
+
+TR::AotAdapter *
+OMR::AotAdapter::self()
+    {
+    return static_cast<TR::AotAdapter *>(this);
+    }
+ void OMR::AotAdapter::initializeAOTClasses(TR::RawAllocator* rawAllocator,TR::CodeCacheManager* cc){
+//   _sharedCache = new (PERSISTENT_NEW) TR::SharedCache("wasm_shared_cache", "/tmp");
+  _reloRuntime = new (PERSISTENT_NEW) TR::SharedCacheRelocationRuntime (NULL);
+  _codeCacheManager = cc;
+ }
+ 
+void OMR::AotAdapter::storeExternalSymbol(const char *symbolName, void* symbolAddress){
+    _reloRuntime->registerLoadedSymbol(symbolName,symbolAddress);
+}
+
+
+void OMR::AotAdapter::storeAOTMethodAndDataInTheCache(const char* methodName){
+    TR::AOTMethodHeader* hdr =_methodNameToHeaderMap[methodName];
+    // _sharedCache->storeEntry(methodName,hdr->serialize(),hdr->sizeOfSerializedVersion());
+}
+
+TR::AOTMethodHeader* OMR::AotAdapter::loadAOTMethodAndDataFromTheCache(const char* methodName)
+{
+    TR::AOTMethodHeader* methodHeader = NULL;
+    void* cacheEntry;
+    // void* cacheEntry = reinterpret_cast<uint8_t*> (_sharedCache->loadEntry(methodName));
+    if (cacheEntry!=NULL){
+        methodHeader = new TR::AOTMethodHeader((uint8_t*)cacheEntry);
+        void* codeStart = methodHeader->compiledCodeStart;
+        _reloRuntime->registerLoadedSymbol(methodName, codeStart);
+        registerAOTMethodHeader(methodName,methodHeader);
+    }
+
+    return methodHeader;
+}
+void OMR::AotAdapter::registerAOTMethodHeader(std::string methodName,TR::AOTMethodHeader* header){
+    _methodNameToHeaderMap[methodName] = header;
+}
+
+TR::RelocationRuntime* OMR::AotAdapter::getRelocationRuntime(){
+    return _reloRuntime->self();
+}
+
+TR::AOTMethodHeader* OMR::AotAdapter::createAndRegisterAOTMethodHeader(const char* methodName, uint8_t* codeStart, uint32_t codeSize,
+						  uint8_t* dataStart, uint32_t dataSize){
+     TR::AOTMethodHeader* hdr = (TR::AOTMethodHeader*)new (TR::AOTMethodHeader) (codeStart,codeSize,dataStart,dataSize);
+     registerAOTMethodHeader(methodName,hdr );
+}
+
+void OMR::AotAdapter::relocateRegisteredMethod(const char* methodName)
+{
+    TR::AOTMethodHeader* hdr =  getRegisteredAOTMethodHeader(methodName);
+    _reloRuntime->self()->prepareRelocateAOTCodeAndData(hdr,hdr->compiledCodeStart);
+}
+
+ TR::AOTMethodHeader* OMR::AotAdapter::getRegisteredAOTMethodHeader(const char* methodName){
+    std::string method(methodName);
+    TR::AOTMethodHeader* result =_methodNameToHeaderMap[methodName];
+    if (result==NULL)
+        {
+        result = loadAOTMethodAndDataFromTheCache(methodName);
+        }
+    return result;
+ }
+void OMR::AotAdapter::storeHeaderForCompiledMethod(const char* methodName)
+{
+    std::string method(methodName);
+
+    if (_methodNameToHeaderMap[methodName] != NULL)
+    {
+       storeAOTMethodAndDataInTheCache(methodName);
+    } else {
+       std::cerr << "Method " << methodName << " not found!" << std::endl;
+    }
+}
+bool isMethodAllocatedAlready(void* pc){
+    return false;
+}
```

Comment:
What is the point of this?

Reply:
This code will be removed soon, it was put there for demonstration purposes, however, I agree, it is meaningless in this context. Thank you for your comment!